### ISSUE TITLE
blades not well integrated into hub?

### ISSUE THREAD

**Issue by Bazmundi** [hyperair/fan-blades#2]
May be my fault but there is a curious thing going on.

I was trying to print [5 blade drmadthrust 50mm](https://hobbyking.com/en_us/hobbyking-50mm-alloy-edf-v2-4200kv-4s-version.html?countrycode=AU&gclid=CjwKCAjwiMj2BRBFEiwAYfTbCvKKiJ5Bh7yTj_UA3dZB_QMxlQsk1Nk9Zv16dhBcesh9ddeyn-asDxoCqQoQAvD_BwE&gclsrc=aw.ds&___store=en_us).

Note the nose cap.  The nose cap is held on by a ridge in the cap that fits snugly into the inner diameter of the hub.  A screw is used to hold the nose cone on.  The hub has an internal platter that I replicated by adding the following code:

```
union() {
  hub(); 
  difference() {
    translate([0,0,2.4]) cylinder(h=2,d=18.9);
    cylinder(h=10,d=5.5);
  }
}
```

However, the default is a design with cylinder for the hub thicker one end than the other.   With the given translation I had the platter at the thicker end.  But the dimension parameters for the hub, being tuned to allow the nose cone to be fitted snugly, meant I ended up with a nose cone that did fit snugly in the end of the hub opposite the platter but not the end the nose cone was intended (nearest to the platter).

Rather than change the translation, I modified the code to the following:
```
union() {
  mcad_tube(hub_od,hub_d, blade_height); 
  difference() {
    translate([0,0,2.4]) cylinder(h=2,d=18.9);
    cylinder(h=10,d=5.5);
  }
}
```

That seemed straight forward enough to do.

The print resulted in a hub that allowed the nose cone to fit snugly either end of the hub as you would expect, if the cylinder wall was consistent.

There was a problem though.  Two previous prints, using stock code, with slightly different parameters for blade diameter (as I was tuning the blade diameter to better match the shroud ID) produced blades well integrated into the hub.  When I modified the code to use  mcad_tube(), I found that I broke off one of the blades because that blade that broke off, and in fact all the others, had a gap between the blade and the hub, along the length of the blade intersection with the hub.  The gap was on both sides of the blade.  The inner edge of the blade was inline with the inner hub wall.  Essentially then, the blade was not actually attached to the hub, except for the point where the blade intersected the platter (so a small section a couple of millimetres squared).

Would this effect be because I am not using tab() anymore?  

Although, it might only be a print phenomenon, since I noted that the wall of hub was .02 mm narrower using mcad_tube(hub_od,hub_d, blade_height) instead of using hub().   So I will reprint with a thicker wall.  I noticed that there was a *0.65 factor for the wall thickness then multiplied later, dependant upon context.  So, as there were no comments in the code it isn't apparent why all the tweaking in the original code.  So I am still muddling through that aspect.

I might at least try changing my translation of the platter to the other end using the original hub().  I am only after linear blades for the application.  

I am otherwise a bit vague on which direction of flow for parabolic blades but my best guess is translating the platter to the other end won't be of much use, since that will be the wrong end for the parabolic blades and how they are constructed currently?

So, were there print anomalies in the original design that the code tweaking was trying to "fix"?

**Comment by hyperair** [comment]
Hi @Bazmundi, this repo was an attempt at approximating replacement fan blades for computer/3d-printer fans such as these, so I'm not sure how well they would perform on a drone motor instead.
![image](https://user-images.githubusercontent.com/88355/83348255-6fba7c80-a35d-11ea-9717-47e05f16ff1b.png)

In any case, it's quite hard to understand the issue you're talking about without pictures and example code, so could you upload some please?

**Comment by hyperair** [comment]
I've made a different set of fan blades at https://github.com/hyperair/naca-fan-blades which might be of more interest to you, by the way. They've got an airfoil profile, so they work a little better than the simple curved plastic strips that are in this implementation.

**Comment by Bazmundi** [comment]
Yep. Found the others thanks. I did also work out the print problem was a print phenomenon  as I thickened the hub and the problem went away.[Image]

Linear in foreground left. Original foreground right. NACA blades in EDF unit. I did go away from Razer sharp trailing edges with a cheat. Thanks for sharing your code!

**Comment by hyperair** [comment]
I think the image might have gotten left out of your previous comment. I just see a placeholder `[Image]` in your text. You'll have to use the Github web-ui or an external hosting service for this, I think.

**Comment by Bazmundi** [comment]
Actually it's for a blimp. But I understand what you are saying about drones. And yes. I may yet still need to find a buyer for my current edf and then buy paired cw/ccw units. I was as fascinated in how your code worked. Some of your ideas may also translate to ribs for wings.

Get Outlook for Android<https://aka.ms/ghei36>
